### ISSUE TITLE
Logger shutdown

### ISSUE THREAD

**Issue by alekseytimonin** [NLog/NLog#2778]
Hi,
I just starting to work with NLog.
I wrote a wrapper for the logger, to unify work with logger for all our applications.
In configuration of the logger I use AsyncWrapper for better performance, and found a recommendation to use LogManager.Shutdown method to ensure that no log prints will be lost when application is shutting down.
I know nothing about the application, that going to use my logger wrapper - it can be web app, executable or any other...

So my questions are:

Is there any recommendation where to call LogManager.Shutdown method? 
Is it good idea to call LogManager.Shutdown in my wrapper finalizer?
Does NLog takes care about it and I don't need to call LogManager.Shutdown method?
Is it any configuration I can use instead of calling LogManager.Shutdown?

Bottom line: What is the right way to do that?

Thanks,
Aleksey

**Comment by snakefoot** [comment]
Only the main application should call `NLog.LogManager.Shutdown()`.

If you have a library/component then you probably have a Close/Dispose-method to stop own timers/threads/etc. In that method then you can call `NLog.LogManager.Flush()`.

NLog has an auto-flush/auto-close on AppDomain-Unload, but it is not super-reliable, as shutdown is tricky business as it will be aborted prematurely if not fast enough to complete the shutdown.

**Comment by 304NotModified** [comment]
I assume your question has been answered, if not, please let us know!

**Comment by dorfire** [comment]
Does it matter if I call `Shutdown()` in the Main() method or in my main class's destructor?

**Comment by snakefoot** [comment]
> Does it matter if I call Shutdown() in the Main() method or in my main class's destructor?

Please create new issue for your own question, since this issue is closed. Btw. i c# then class destructor usually means finalizer and those are not guaranteed to be called. Better to use Dipose or in finally in main-method.
